src/call_wrapper.ts

```diff
@@ -23,25 +23,24 @@ class CallWrapper {
         return fixedParameters;
     }
 
-    private static readMemory(parameters: any[], fixedParameters: any[], readableMemory: Uint32Array) {
-        let index = 0;
-
-        for (let i = 0; i < parameters.length; i++) {
-            if (parameters[i] instanceof Array) {
-                const array = parameters[i];
+    private static readMemory(parameters: any[],
+                              outParameters: any[],
+                              fixedParameters: any[],
+                              readableMemory: Uint32Array) {
 
-                for (let j = 0; j < array.length; j++) {
-                    array[j] = readableMemory[fixedParameters[index] / 4 + j];
-                }
+        for (const outParameter of outParameters) {
+            const parameterIndex = parameters.indexOf(outParameter);
+            const array = parameters[parameterIndex];
```

Comment:
This should be changed to `outArray` or something similar

Reply:
And the index from above could also be `-1` if it doesn't exist in the `parameters` array. This should be checked and an error should get thrown.